ocs_ci/utility/kms.py

```diff
@@ -0,0 +1,388 @@
+        with open(vault_hcl.name, "w") as hcl:
+            hcl.write(policy)
+
+        if not config.ENV_DATA.get("VAULT_POLICY"):
+            self.vault_policy_name = (
+                f"{constants.VAULT_DEFAULT_POLICY_PREFIX}-" f"{self.cluster_id}"
+            )
+        else:
+            self.vault_policy_name = config.ENV_DATA.get("VAULT_POLICY")
+
+        cmd = f"vault policy write {self.vault_policy_name} {vault_hcl.name}"
+        out = subprocess.check_output(shlex.split(cmd))
+        if "Success" in out.decode():
+            logger.info(f"vault policy {self.vault_policy_name} created")
+        else:
+            raise VaultPolicyCreationFailed(
+                f"Failed to create policy f{self.vault_policy_name}"
+            )
+        self.vault_path_token = self.generate_vault_token()
+
+    def generate_vault_token(self):
+        """
+        Generate a token for self.vault_policy_name
+
+        Returns:
+            str: vault token
+
+        """
+        cmd = f"vault token create -policy={self.vault_policy_name} " f"--format=json"
+        out = subprocess.check_output(shlex.split(cmd))
+        json_out = json.loads(out)
+        return json_out["auth"]["client_token"]
+
+    def get_cluster_id(self):
+        """
+        Get cluster UUID
+        Not relying on metadata.json as user sometimes want to run
+        only with kubeconfig for some tests
+
+        Returns:
+            str: cluster UUID
+
+        """
+        cluster_id = run_cmd(
+            "oc get clusterversion version -o jsonpath='{.spec.clusterID}'"
+        )
+        return cluster_id
+
+    def get_vault_cli(self):
+        """
+        Download vault based on platform
+        basically for CLI purpose
+
+        """
+        if distro.linux_distribution == "CentOS Linux":
+            deps = " ".join(constants.VAULT_CENTOS_DEPS)
+            cmd = f"sudo yum install -y {deps} "
+            run_cmd(cmd)
+            cmd = (
+                f"sudo yum-config-manager --add-repo " f"{constants.VAULT_CENTOS_REPO}"
+            )
+            run_cmd(cmd)
+            cmd = "sudo yum -y install vault"
+            run_cmd(cmd)
+        else:
+            raise (
+                VaultPlatformNotSupported,
+                "Vault CLI for this platform not supported",
+            )
+
+    def deploy_vault_internal(self):
+        """
+        This function takes care of deployment and configuration for
+        internal mode vault deployment on OCP
+
+        Raises:
+            exceptions.FailedVaultDeployment
+
+        """
+        pass
+
+    def gather_vault_config(self):
+        """
+        This function populates the vault configuration
+
+        """
+        if self.vault_deploy_mode == "external":
+            vault_conf = load_auth_config()["vault"]
+            return vault_conf
+
+
+kms_map = {"vault": Vault}
+
+
+def get_kms_deployment():
+    provider = config.DEPLOYMENT["kms_provider"]
```

Comment:
Raise some specific exception about not supported KMS Provider if keyError

Reply:
@petr-balogh ack, done in the next commit update